source/utils/types.ts

```diff
@@ -113,10 +119,27 @@ export interface Delays {
 	request?: number;
 }
 
-export type Headers = Record<string, string | string[]>;
+export type Headers = Record<string, string | string[] | null | undefined>;
```

Comment:
```suggestion
export type Headers = Record<string, string | string[] | undefined>;
```

We decided to only allow `undefined` to clear headers and not `null`.

Reply:
This will need to update the test on removing null headers - should I also do it?

Reply:
Yes, thanks :)